ion-hash/tests/ion_hash_tests.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+use ion_hash::{self, IonHashDigest, IonHasher};
+use ion_rs::result::IonResult;
+use ion_rs::value::loader::{loader, Loader};
+use ion_rs::value::*;
+use std::{
+    fs::read,
+    sync::{Arc, Mutex},
+};
+
+#[derive(Default, Clone)]
+struct TestDigest {
+    inner: Arc<Mutex<TestDigestInner>>,
+}
+
+#[derive(Default)]
+struct TestDigestInner {
+    updates: Vec<u8>,
+    digest: Vec<u8>,
+}
+
+impl TestDigest {
+    fn inspect(&self) -> (Vec<u8>, Vec<u8>) {
+        let inner = self.inner.lock().unwrap();
+        (inner.updates.clone(), inner.digest.clone())
+    }
+}
+
+impl IonHashDigest for TestDigest {
+    fn update(&mut self, bytes: impl AsRef<[u8]>) {
+        let mut inner = self.inner.lock().unwrap();
+        for b in bytes.as_ref() {
+            inner.updates.push(*b);
+            inner.digest.push(*b);
+        }
+    }
+
+    fn finalize(self) -> Vec<u8> {
+        let inner = self.inner.lock().unwrap();
+        inner.digest.clone()
+    }
+}
```

Comment:
I really need some help with explaining why these types are here--my read below makes me think you are doing some kind of mocking or internal aggregation, but I am not sure.  Either way I didn't read any concurrent code below so is that just there in case you do make it multi-threaded or is it just `RefCell`?

Reply:
The latter :)